### ISSUE TITLE
"Comments" refactoring can lead to compile errors

### ISSUE THREAD

**Issue by martinschemel** [JnRouvignac/AutoRefactor#149]
Used AutoRefactor Version: 1.0.0

## Before Refactoring
There are two fields in a java class:

```java
	/** We use a regex because we don't care for the display of the duration **/
	private static final String REGEX = "${Text}[ \\(\\d\\d:\\d\\d:\\d\\d\\)]*"; // machtes e.g. "<whateverText> (00:00:04)" and "<whateverText>"

	// *********** result view node texts without time ***********//
	private static final String TS1 = "Test Set: ts1";
```

### Actual

```java
	/** We use a regex because we don't care for the display of the duration *.*/
	private static final String REGEX = "${Text}[ \\(\\d\\d:\\d\\d:\\d\\d\\)]*"; /**
* Machtes e.g. "<whateverText> (00:00:04)" and "<whateverText>"

	 * *********** result view node texts without time ***********//
	 */
	private static final String TS1 = "Test Set: ts1";
```
### Expectation (maybe not 100% correct)

```java
	/** We use a regex because we don't care for the display of the duration. **/
	private static final String REGEX = "${Text}[ \\(\\d\\d:\\d\\d:\\d\\d\\)]*"; // machtes e.g. "<whateverText> (00:00:04)" and "<whateverText>"

	// *********** result view node texts without time ***********//
	private static final String TS1 = "Test Set: ts1";
```
Please fix.

**Comment by martinschemel** [comment]
Note sure if this is a duplicate of https://github.com/JnRouvignac/AutoRefactor/issues/80

**Comment by JnRouvignac** [comment]
Hi Martin,

And thanks for reporting this.
I agree it should not be refactored like this.
As a developer I would rather expect an output like this:

```java
    /**
     * We use a regex because we don't care for the display of the duration.
     * Machtes e.g. "<whateverText> (00:00:04)" and "<whateverText>"
     */
    private static final String REGEX = "${Text}[ \\(\\d\\d:\\d\\d:\\d\\d\\)]*"; /**

    /** *********** result view node texts without time ************/
    private static final String TS1 = "Test Set: ts1";
```

Now implementation wise this will require more work :)

**Comment by martinschemel** [comment]
Hi Jean,

your suggestion looks even more satisfying (except the "/**" at the end of the REGEX field line which I guess is just a typo). I think I personally would like it this way. 

On the other hand I could imagine that one writes a usual inline comment ("//") on purpose because he doesn't want it to be part of the actual java doc. The same applies for the "TS1" field. Maybe someone used "//" instead of "/**" on purpose. So the "comments" refactoring would perform semantic changes which maybe shouldn't be the case. But it's your decision respectively I did not spend as much as thoughts as you into this stuff ;)

**Comment by JnRouvignac** [comment]
> (except the "/**" at the end of the REGEX field line which I guess is just a typo)

Yes, it is a copy / paste typo. I'll fix my previous comment so it does not look too weird to a reader.

> On the other hand I could imagine that one writes a usual inline comment ("//") on purpose because he doesn't want it to be part of the actual java doc.

For a field (line comments on end of line are only used for fields) I cannot think of a good reason. Most of the time java programmers are making fields private so javadoc is not generated. However such comments escape in-IDE javadocs while browsing the code, which reduces their usefulness.

For TS, the case is even simpler in my opinion: there is no javadoc, so this should actually be a javadoc.

**Comment by JnRouvignac** [comment]
Compile error is now fixed.
This fix will be part of AutoRefactor v1.1.0.

Issue #152 tracks "Attach line comments on fields to existing javadocs".
